playbooks/compliance_check.yaml

```diff
@@ -4,35 +4,33 @@
   tasks:
     - name: Run compliance script
       ansible.builtin.shell: |
-        python3 ~/Tests/scs-compliance-check.py ~/Tests/scs-compatible-iaas.yaml -s {{ cloud }} -a os_cloud={{ cloud }} -o report.yaml -C
-      register: result
+        python3 ~/Tests/cleanup.py -c {{ item }} --prefix _scs-
+        python3 ~/Tests/scs-compliance-check.py ~/Tests/scs-compatible-iaas.yaml -s {{ item }} -a os_cloud={{ item }} -o report-{{ item }}.yaml -C
+        ssh-keygen -Y sign -f ~/id_subject -n report report-{{ item }}.yaml
+        curl --data-binary @report-{{ item }}.yaml.sig --data-binary @report-{{ item }}.yaml -H "Content-Type: application/x-signed-yaml" -H "Authorization: Basic $auth" https://compliance.sovereignit.cloud/reports
```

Comment:
Is there a specific technical reason to put all these steps into on task? (Maybe to avoid repeated `loop` setups?) Because before, this was probably a bit more user friendly to have separate tasks for the cleanup, the actual check and the submission to the compliance monitor.

If the loop setup is the reason, maybe you can put the tasks into a separate file, e.g., `compliance_check_steps.yaml` and include that via `include_tasks` (there is also the `block` construct in ansible, but I had to look up if you can loop over blocks and apparently it is not possible, e.g., see [this blog post](https://ericsysmin.com/2019/06/20/how-to-loop-blocks-of-code-in-ansible/)).

Reply:
Well, there are reasons, I don't know if they are technical.

1. I dislike Ansible. It's a bunch of peculiar, badly documented yaml that reinvents a lot of wheels just to claim that it's declarative, which doesn't make a lot of sense in this case.
2. I would like to make this whole loop run concurrently, and I think this might be possible with `async` (but then I would have to specify how long I expect this to take??) or I just use the tried and true shell (with `&` and `wait`).
3. Maybe there are more...

Reply:
BTW: funny, because I had found the same blog post when I asked myself the same question :)